crates/api_common/src/utils.rs

```diff
@@ -293,6 +293,20 @@ pub async fn check_private_instance(
   Ok(())
 }
 
+#[tracing::instrument(skip_all)]
+pub async fn is_private_instance(
+  pool: &DbPool
+) -> Result<bool, LemmyError> {
+  let site = blocking(pool, Site::read_local_site).await?;
+    // The site might not be set up yet
+    if let Ok(site) = site {
+      if site.private_instance {
+        return Ok(true);
+      }
+    }
+    return Ok(false);
+}
+
```

Comment:
This helper function isnt helping much, and is only used in one place. I would simply move this logic into images.rs directly.